#388, #393 - LogsViewer - Button in log record for filtering/quarantining similar logs

przemekxa opened the pull request:
## Description

Added two buttons in expanded log record:
- "Filter similar logs"
- "Add similar logs to quarantine"

They open appropriate windows (adding a log or adding a quarantine rule). The contents of the header of the last variable data is then set as the regular expression.

## Motivation and Context

#388 and #393

## Screenshots

![Zrzut ekranu 2021-12-31 o 15 56 26](https://user-images.githubusercontent.com/31923361/147829583-cf94c772-fa84-44fa-a100-27e4fca9d41d.png)

![Zrzut ekranu 2021-12-31 o 15 56 38](https://user-images.githubusercontent.com/31923361/147829590-d0b68cc3-9c45-413b-ba4b-d520cec656bd.png)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code style](https://github.com/wttech/cogboard/blob/master/CONTRIBUTING.md#coding-conventions) of this project.
- [x] My change requires a change to the documentation.
- [ ] Automated functional tests have been added or modified to cover my changes (if applicable)
- [ ] I have updated the documentation accordingly.

---

I hereby agree to the terms of the Cogboard Contributor License Agreement.

janeksmielowski commented:
spójrz proszę na prototyp w docu CogBoard Plan. Chcielibyśmy mieć dwa guziczki z ikonkami do tych operacji, bez tekstu

przemekxa commented:
Changed the way the buttons look, added tooltips (one of them is visible on the screenshot):

![Zrzut ekranu 2022-01-3 o 20 22 06](https://user-images.githubusercontent.com/31923361/147972147-8df9be92-dd35-48c5-8b10-9bd12e63fae7.png)

janeksmielowski commented on cogboard-webapp/src/components/widgets/types/LogViewerWidget/Toolbar/FilterPicker/index.js:
```diff
@@ -97,6 +103,8 @@ const FilterPicker = ({ widgetLocalStorage, wid, quarantine }) => {
         widgetLocalStorage={widgetLocalStorage}
         wid={wid}
         quarantine={quarantine}
+        filterSimilarLogsState={filterSimilarLogsState}
```
jakieś niebezpieczne rzeczy tutaj się dzieją - przekazywanie stanu + settera w dół kilka poziomów. Może jakiś kontekst by tutaj pasował? Po co nam getter i setter na każdym poziomie?
